### PR TITLE
Update Release Notes|News|Download Links|versions.json for v3.5.4

### PR DESCRIPTION
**Pull request by LuciferYang**
image
image
https://issues.apache.org/jira/projects/SPARK/versions/12355093

### CODE REVIEW

**Comment by LuciferYang** [comment]
cc @dongjoon-hyun @HyukjinKwon @yaooqinn @cloud-fan

File: site/documentation.html [hunk 1]
```diff
@@ -158,6 +158,7 @@ <h2><span class="text-capitalize">Apache Spark<span class="tm">&trade;</span></s
 <p>Setup instructions, programming guides, and other documentation are available for each stable version of Spark below:</p>
 
 <ul>
+  <li><a href="/docs/3.5.4/">Spark 3.5.4</a></li>
```
Comment (dongjoon-hyun) [inline site/documentation.html:161]: This seems not working so far. Could you double-check, @LuciferYang ?
- https://spark.apache.org/docs/3.5.4/

Comment (LuciferYang) [reply site/documentation.html:161]: Let me check it

**Comment by dongjoon-hyun** [comment]
Although we have this correctly, it seems that we are hitting the resources issue again.
- https://github.com/apache/spark-website/tree/asf-site/site/docs/3.5.4

Previously, we hit the infra resource issues and decided to redirect old websites to the archive locations instead of here.
- https://lists.apache.org/thread/q9p2nj5x0z9p6cy32c9vd9lo43v4qxls

File: site/documentation.html (hunk 1 above)
Comment (LuciferYang) [reply site/documentation.html:161]: image

It's a bit strange, it works fine on my local machine.

Comment (dongjoon-hyun) [reply site/documentation.html:161]: Yes, it seems to be ASF Infra side publishing issue.

**Comment by dongjoon-hyun** [comment]
To @yaooqinn , do you happen to know how to check ASF website publish step?

**Comment by LuciferYang** [comment]
> To @yaooqinn , do you happen to know how to check ASF website publish step?

might take some time. It's already past midnight in Beijing time, @yaooqinn  should already be resting.

**Comment by dongjoon-hyun** [comment]
I checked ASF website.
- https://infra.apache.org/project-site.html#sitemanagement

I found that the staging is correct.
- https://spark.staged.apache.org/docs/3.5.4/

So, let's merge this in order to trigger the ASF publishing step once more, @LuciferYang .

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Comment by LuciferYang** [comment]
Thank you @dongjoon-hyun

**Comment by dongjoon-hyun** [comment]
It's working now. I verified.

Screenshot 2024-12-20 at 09 01 09

**Comment by LuciferYang** [comment]
Great !!! Thank you once again @dongjoon-hyun

**Comment by dongjoon-hyun** [comment]
BTW,  you can announce Apache Spark 3.5.4 without waiting your Docker publishing.
- https://github.com/apache/spark-docker/pull/77

**Comment by LuciferYang** [comment]
> BTW, you can announce Apache Spark 3.5.4 without waiting your Docker publishing.
> 
> * [Publish 3.5.4 to docker registry spark-docker#77](https://github.com/apache/spark-docker/pull/77)

OK ~

**Pull request merged**